src/components/Table/utils/TableState.ts

```diff
@@ -221,8 +221,31 @@ export class TableState {
       this.columns = columns;
       this.visibleColumnsStorageKey = visibleColumnsStorageKey ?? camelCase(columns.map((c) => c.id).join());
       this.visibleColumns.replace(readOrSetLocalVisibleColumnState(columns, this.visibleColumnsStorageKey));
-      const expandedColumnIds = columns.filter((c) => c.initExpanded).map((c) => c.id);
-      if (isInitial) this.expandedColumns.replace(expandedColumnIds);
+      // list of local storage columns
+      const localStorageColumns = this.persistCollapse ? readExpandedColumnsStorage(this.persistCollapse) : undefined;
+      const expandedColumnIds: string[] = [];
+
+      columns.forEach((c) => {
+        // looks at initExpanded and reads localStorage
+        if (isInitial && c.initExpanded) {
+          expandedColumnIds.push(c.id);
+          expandedColumnIds.concat(localStorageColumns!);
+          this.expandedColumns.replace(expandedColumnIds);
+        } else {
+          // subsequent load, ignoring initExpanded
+          // if its a new column or existing column then create new expanded column ids
+          // new expanded column ids merge with local storage
+          const newExpandedColumnIds = this.columns.filter((col) => col.initExpanded).map((c) => c.id);
```

Comment:
I don't think I understand this 100%.. I searched through `this.columns` and cannot find anything that would help me determine if the column is new or existing..

Reply:
What you'd want to do here is not yet set `this.columns = columns;`. Do that as like a last step. Then you should be able to look at `columns` and see which of the columns do not yet exist in `this.columns` in order to determine which are new.